File: app/src/main/java/com/example/nshutiplanner/unity/BridgeResponse.kt
package com.example.nshutiplanner.unity

import org.json.JSONObject

data class BridgeResponse(
    val success: Boolean = false,
    val latitude: Double = 0.0,
    val longitude: Double = 0.0,
    val displayName: String = "",
    val error: String = ""
) {
    fun toJson(): String {
        return JSONObject().apply {
            put("success", success)
            put("latitude", latitude)
            put("longitude", longitude)
            put("displayName", displayName)
            put("error", error)
        }.toString()
    }

    companion object {
        fun fromJson(json: String): BridgeResponse {
            val obj = JSONObject(json)
            return BridgeResponse(
                success = obj.optBoolean("success", false),
                latitude = obj.optDouble("latitude", 0.0),
                longitude = obj.optDouble("longitude", 0.0),
                displayName = obj.optString("displayName", ""),
                error = obj.optString("error", "")
            )
        }
    }
}

File: Assets/Scripts/BridgeResponse.cs
using System;

// BridgeResponse mirrors the Kotlin BridgeResponse data class in
// app/src/main/java/com/example/nshutiplanner/unity/BridgeResponse.kt.
//
// JSON schema:
// {
//   "success":     bool,
//   "latitude":    double,
//   "longitude":   double,
//   "displayName": string,
//   "error":       string
// }
//
// Field rules:
//   success = true  → error is empty; latitude/longitude are valid; displayName is non-empty.
//   success = false → error is non-empty; latitude/longitude/displayName are zero/empty.
//
// Deserialized via JsonUtility.FromJson<BridgeResponse>(json).
[Serializable]
public class BridgeResponse
{
    public bool success;
    public double latitude;
    public double longitude;
    public string displayName;
    public string error;
}
